wrangler-core/src/main/java/io/cdap/directives/column/CleanseColumnNames.java

```diff
@@ -79,6 +84,7 @@ public List<Row> execute(List<Row> rows, ExecutorContext context) throws Directi
         column = column.toLowerCase();
         // Filtering unwanted characters
         column = column.replaceAll("[^a-zA-Z0-9_]", "_");
+        cleansedColumnNames.put(row.getColumn(i), column);
```

Comment:
we should not depend on row data. All we need is input schema to generate output schema. I believe we can apply the same column transformations on input schema to get the output schema.

Reply:
Done